### PR TITLE
Replace ogg vorbis encoding with flac

### PR DESCRIPTION
**Pull request by JoaCHIP**
Encoding to ogg vorbis reduces the audio quality. Using flac as the intermediate format retains 100% audio quality.

### CODE REVIEW

**Comment by rsxdalv** [comment]
Hi, thank you for the PR!
I have two immediate things I would like to test (but can't right now): does the metadata still get encoded and is the file size as small as it was before.
What I'm thinking is - if there's any difficulty with those two points or perhaps there are separate goals to achieve, this could be another plugin - it could generate flac or ogg (or mp3 since people like the name).
Here's a question - does your use case require the metadata about generation to be present in the file?

**Comment by JoaCHIP** [comment]
The meta data that was stored in the ogg file is now stored in the .flac file instead. I've double checked using both foobar2000 and a hex viewer. Flac files are larger than ogg files. But considering that the result is also stored as a .wav file (twice the size of the flac file), and today's harddrive sizes, I don't think spending that extra space is much of an issue. I'd much rather have full quality. Saving space could be done by not storing a .wav version, but have the browser play the flac file directly. Both firefox and MS edge support flac.

So in short: Yes the meta data is stored in the flac files, but it's not something I really need.

**Comment by rsxdalv** [comment]
> The meta data that was stored in the ogg file is now stored in the .flac file instead. I've double checked using both foobar2000 and a hex viewer. Flac files are larger than ogg files. But considering that the result is also stored as a .wav file (twice the size of the flac file), and today's harddrive sizes, I don't think spending that extra space is much of an issue. I'd much rather have full quality. Saving space could be done by not storing a .wav version, but have the browser play the flac file directly. Both firefox and MS edge support flac.
> 
> So in short: Yes the meta data is stored in the flac files, but it's not something I really need.
> 
> PS: A plugin that generates a flac file is useless if the intermediate file is in a lossy format like ogg. My goal here (as a musician) is to retain full quality, and avoid encoding using lossy formats like ogg/mp3 etc. Flac is 100% quality.

It would generate FLAC from wav. I have some things that depend on the metadata, and it's good news if it works the same or almost the same.
With a few additional changes I can make it a standalone plugin that generates FLAC from wav. I reckon FLAC could be the ultimate format that is used, I imagine that there are ways to convert it to WAV when absolutely necessary (for some old Android phone browser etc.)

**Comment by JoaCHIP** [comment]
The meta data still works!

**Comment by rsxdalv** [comment]
I added some changes (it's both surprisingly difficult and possible to write code on a mobile phone).
If it was done this way, we could have them side by side.
Naturally, it would be better to have the code be flexible (like generate both ogg and flac, only flac or only ogg etc), but that requires a place for settings etc.
With this - the users could by default get _everything_ and then if they didn't want something they could just delete that .py file.
Once I have a better chance I would like to make a more versatile plugin system.

**Comment by rsxdalv** [comment]
> Now, I have to admit that I'm not a python developer. I'm mostly a C#/C++/php guy, so I can't tell if your code is gonna work or not.
> 
> My interpretation of things were that things were always being compressed to ogg internally, and then that got turned into .wav. If I'm wrong aaf,fvqbout that, this whole merge request can be thrown away. And in that case I've been wasting your time, which would be pretty embarassing.

I think the code makes it seem like it gets converted to ogg prematurely, but that's just my spaghetti. Actually the files.get("ogg") is just the filename that, for no logical reason, gets passed into the plugin. (probably because before it was a plugin, the code was just all together in one file) The actual WAV to ogg happens purely within the plugin.

I didn't know if FLAC was going to work, so your work has probably improved the scaling of this project since wavs are so large.

Once I have access to a workstation I will modify this to fit as I mentioned.

If you have any more questions or needs code-wise let me know!

Thank you for taking the time with this project.

**Comment by JoaCHIP** [comment]
I'm the one saying thanks. In fact, I've examined the .wav vs. ogg files more closely now, and you're right. The .wav files do indeed contain **full quality audio**, which means my whole pull request is not necessary at all.

Now that I know the .wav files are indeed full quality, you can safely delete / reject my whole merge request. I feel like I owe you a beer or something. 😅
